determined-ai/determined
# fix: make agents summary more accurate under k8s with multiple RPs

## Description

Uses the original all-cluster summary in the `/agents` endpoint when there's one pool and it has no quotas. Otherwise uses quotas to come up with a slots analog.

## Test Plan

Outside of k8s, the cluster page should be the same as before. To test the changes under k8s, this is what I did:
- (Using the instructions found in `determined/tools/k8s`) run a k8s version of Determined with no extra namespaces/resource pools configured and no quotas. The cluster info page should report the whole cluster's hardware.
- Add a quota for that one namespace/resource pool. The cluster info page should reflect what's allowed in the quota.
- Add another namespace, corresponding resource pool, and quota. All resource pools should still reflect what's available to them through their quota.

## Commentary (optional)

- The slots' containers in the `/agents` endpoint code feels awkward, but if callers of the endpoint wanted to use it to total up how many containers were running, this implementation should make that possible.

## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Tickets
DET-8714, DET-8809
